junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/SuiteTestDescriptor.java

```diff
@@ -148,20 +149,25 @@ private static String getSuiteDisplayName(Class<?> testClass) {
 		// @formatter:on
 	}
 
-	void execute(EngineExecutionListener parentEngineExecutionListener,
-			NamespacedHierarchicalStore<Namespace> requestLevelStore) {
-		parentEngineExecutionListener.executionStarted(this);
+	void execute(EngineExecutionListener executionListener, NamespacedHierarchicalStore<Namespace> requestLevelStore,
+			CancellationToken cancellationToken) {
+
+		if (cancellationToken.isCancellationRequested()) {
+			executionListener.executionSkipped(this, "Execution cancelled");
+		}
```

Comment:
Neat implementation.

Reply:
Actually, there was a `return` missing there. 😳

Reply:
Good catch. I read the code as a fall-through was intended.